AFK team mate v2: better ship handling + tests + bugfix

VariableVince opened the pull request:
## Description:

See PR https://github.com/openfrontio/OpenFrontIO/pull/2203

It was reverted. This unreverts it, with an added fix for boat troops not getting returned to owner. And small comment updates. And a const for boatOwner to re-use.

The added bugfix is check for this.sourceTile === null in the retreat() function in AttackExecution. A boat attack always sets removeTroops to false because the troops were already removed from owner troops when the boat departed. They don't have to be removed again in AttackExecution init, when the boat lands and the attack starts. But at the end of the attack, in retreat() in AttackExecution, the starting/boat troops still need to be returned to the owner. That's why even if removeTroops is false, when sourceTile is not null (only when it's a boat attack) we add back the troops to the owner.

## Please complete the following:

- [x] I have added screenshots for all UI updates
- [x] I process any text displayed to the user through translateText() and I've added it to the en.json file
- [x] I have added relevant tests to the test directory
- [x] I confirm I have thoroughly tested these changes and take full responsibility for any bugs introduced

## Please put your Discord username so you can be contacted if a bug or regression is found:

tryout33

VariableVince committed:
All together now

VariableVince committed:
boat fix and small comment update

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/openfrontio/OpenFrontIO?pullRequest=2396) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/openfrontio/OpenFrontIO?pullRequest=2396) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/openfrontio/OpenFrontIO?pullRequest=2396) it.</sub>

VariableVince committed:
Fix

VariableVince committed:
Fix comment

VariableVince committed:
Final fix

VariableVince commented on src/core/execution/TransportShipExecution.ts:
```diff
@@ -173,11 +177,43 @@ export class TransportShipExecution implements Execution {
+    ) {
+      this.attacker = this.boat.owner();
+      this.originalOwner = this.boat.owner(); // for when this owner disconnects too
+    }
```
This statement is incorrect "If an enemy (or any non‑teammate) conquers the disconnected player, captureUnit still hands them the boat" since non-teammates do not capture transport ships or warships. When a non-teammate conquers a (disconnected) player, the ships are deleted ('boated into a sea mine'), like they always were. So it is correct that this block never fires in that case because it won't happen.

What is a good suggestion is to use const = boatowner so i'll change that.

VariableVince committed:
Const boatowner

FxMorin committed:
Performance: Remove un-optimized blurring (#2466)

## Description:
I haven't been able to play openfront for a while now, so today I took
some time to fix the performance issue.
It turns out its `backdrop-filter: blur(5px);` being used on a
background element, causing the entire website to re-paint on nearly
every change.
This causes the game to run at 8 fps on my computer, with chrome
reporting 700ms of INP presentation delay.

The solution here was simply to pre-blur the background image.

<details>
<summary>Here's the instructions on how to pre-blur exactly like it
currently does</summary>

Install sharp
`npm install sharp`  

Run blur
```ts
sharp(`./resources/images/EuropeBackground.webp`)
  .blur(5)
  .toFile(`./resources/images/EuropeBackgroundBlurred.webp`);
```
This could be automated if you plan to do more backgrounds.  
</details>
  

The surprising part is that I'm running a A5000 with a Ryzen 9 5950X. So
I usually never lag from anything, which is what made this kinda
interesting. I assume the issue is caused by my 4K display or Kubuntu.

## Video:
This video showcases the 700+ms of delay on the main menu buttons. Video
is in real-time.


https://github.com/user-attachments/assets/ad5ff509-6bc7-48ff-8cba-85409be774f0



## Please complete the following:

- [x] I have added screenshots for all UI updates
- [x] I process any text displayed to the user through translateText()
and I've added it to the en.json file
- [x] I have added relevant tests to the test directory
- [x] I confirm I have thoroughly tested these changes and take full
responsibility for any bugs introduced

<details>
<summary>Discord Username</summary>

fx.morin

</details>

VariableVince committed:
Merge branch 'v26' into afk-changes-withboatfix

evanpelle reviewed (dismissed):
thanks!

VariableVince committed:
Merge branch 'main' into afk-changes-withboatfix

ryanbarlow97 committed:
Merge branch 'main' into afk-changes-withboatfix

evanpelle reviewed (approved):
Thanks!
